src/middleware/jira-symmetric-jwt-middleware.test.ts

```diff
@@ -0,0 +1,172 @@
+
+describe("jiraSymmetricJwtMiddleware", () => {
+	let app: Application;
+	let locals;
+	let session;
+
+	beforeEach(async () => {
+
+		app = createApp();
+
+		await Installation.install({
+			clientKey: "jira-client-key",
+			host: jiraHost,
+			sharedSecret: testSharedSecret
+		});
+
+		when(booleanFlag).calledWith(
+			BooleanFlags.NEW_JWT_VALIDATION,
+			expect.anything()
+		).mockResolvedValue(true);
+
+	});
+
+	it("should throw error when token is missing and no jiraHost in session", async () => {
+
+		await supertest(app)
+			.get(`/test`)
+			.then((res) => {
+				expect(res.status).toEqual(401);
+				expect(res.text).toEqual("Unauthorised");
+			});
+	});
+
+	it("should throw error when issuer is missing", async () => {
+
+		const token = getToken(testSharedSecret, "");
+
+		await supertest(app)
+			.get(`/test`)
+			.query({ jwt: token })
+			.then((res) => {
+				expect(res.status).toEqual(401);
+				expect(res.text).toEqual("Unauthorised");
+			});
+	});
+
+	it("should throw error when no installation is found", async () => {
+
+		const token = getToken(testSharedSecret, "jira-wrong-key");
+
+		await supertest(app)
+			.get(`/test`)
+			.query({ jwt: token })
+			.then((res) => {
+				expect(res.status).toEqual(401);
+				expect(res.text).toEqual("Unauthorised");
+			});
+	});
+
+	it("should return valid response when token is valid", async () => {
+
+		const token = getToken(testSharedSecret);
+
+		await supertest(app)
+			.get(`/test`)
+			.query({ jwt: token })
+			.expect(200);
+	});
+
+	it("should set res.locals and req.session when token is valid", async () => {
+
+		locals = {};
+		session = {};
+		const token = getToken(testSharedSecret);
+
+		await supertest(app)
+			.get(`/test`)
+			.query({ jwt: token })
+			.then((res) => {
+				expect(res.status).toEqual(200);
+				expect(locals.installation.jiraHost).toEqual(jiraHost);
+				expect(locals.jiraHost).toEqual(jiraHost);
+				expect(session.jiraHost).toEqual(jiraHost);
+			});
+	});
+
+	it("should throw error when token secret mismatch", async () => {
+
+		const token = getToken("wrong-secret");
+
+		await supertest(app)
+			.get(`/test`)
+			.query({ jwt: token })
+			.then((res) => {
+				expect(res.status).toEqual(401);
+				expect(res.text).toEqual("Unauthorised");
+			});
+	});
+
+	it("should return valid response when jiraHost set in session", async () => {
+
+		session.jiraHost = jiraHost;
+
+		await supertest(app)
+			.get(`/test`)
+			.expect(200);
+	});
+
+	it("should set res.locals and req.session when jiraHost set in session", async () => {
+
+		locals = {};
+		session.jiraHost = jiraHost;
+
+		await supertest(app)
+			.get(`/test`)
+			.then((res) => {
+				expect(res.status).toEqual(200);
+				expect(locals.installation.jiraHost).toEqual(jiraHost);
+				expect(locals.jiraHost).toEqual(jiraHost);
+				expect(session.jiraHost).toEqual(jiraHost);
+			});
+	});
+
+	it("should throw error when invalid jiraHost set in session", async () => {
```

Comment:
nit: In this case, should we also drop the value from session? E.g. let's say for some reason the value got malformed. In this case, if we are not removing it from the session, the user will be stuck...